### DISCUSSION TITLE
DDNS

### DISCUSSION THREAD

**Discussion by HPMSGEN8** [qdm12/gluetun#1206] [Q&A]
Good Day

So I installed this on my docker and I have it running and my other containers are working though it so that is good.

Now how do I setup a DDNS address since the VPN IP Changes and I am using Expresss VPN.

What is a good service to use as well.

Thanks

**Comment by bnhf** [comment]
@HPMSGEN8 

DDNS is only required when you're setting up your own VPN server and you're getting a Dynamic DNS address from your ISP. in this case you're running a VPN client, so you'll be using hostnames, country names or city names as described in the Gluetun ExpressVPN Wiki:

https://github.com/qdm12/gluetun/wiki/ExpressVPN

You shouldn't be using IP addresses directly -- use the hostname if you want  specific server. Otherwise, specify the city or country you want as your exit point.

**Reply by HPMSGEN8** [reply]
Thanks @bnhf 

The VPN IP changes all the time based on the country I selected.

So if I want to use DDNS Service I still need to use a hostname.

What service can I use with docker as I read that cloudflare does not allow Plex and other things so what is recommend.

Thanks

**Reply by bnhf** [reply]
@HPMSGEN8 

I follow you now, you're talking about your public IP changing through the VPN.

I like DuckDNS myself, as it's free for up to 5 DNS records. You'll also need a way to update the record with your latest and greatest IP automatically.  I see the developer of gluetun has another project that should do just that, and it's containerized. Setup his DDNS Updater to connect through gluetun, and set it up to update your DuckDNS account.

You mentioned Plex, and that's a slightly different animal. The Plex Media Server (PMS) will detect your public IP itself, and then test that you have port forwarding configured.

To do this through a VPN provider, you'll need to login to your VPN account and follow the instructions. That same port will need to be opened through the gluetun firewall. So Plex uses your public IP directly -- no DDNS required.

Plex is fussy about "double NAT" though, so you'll have to see how it likes your setup when you verify remote access through PMS.

**Reply by HPMSGEN8** [reply]
Thanks a lot @bnhf 

Yea the VPN.

Plex as just an example of a service as I have read that someone providers do not like it if you use "Services"

Let me give this all a try.

Thanks again
